### ISSUE TITLE
Events not published correctly over channels after v3 to v4 migration

### ISSUE THREAD

**Issue by wgley3** [feathersjs/feathers#1641]
We have just migrated to feathers v4 from v3.

We had a working chat room feature, but now, new service creation events (corresponding to new messages) are not being appropriately published on our custom channels:

```
app.service('messages').publish('created', (data, context) => {
    console.log('messages created event fired')
    console.log(app.channel(`convo/${data.conversationId}`).connections)
    return [app.channel(`convo/${data.conversationId}`)];
  });
```
When we create a new message, this code block is triggered and the console statements here indicate that there is at least one active connection in the `conversation` corresponding to the `conversationId`. Yet the client doesn't receive any new message events. 

Another clue is that when I change the code to leave users in the `anonymous` channel and publish this event to the `anonymous` channel, the client _does_ receive the update appropriately:

```
app.service('messages').publish('created', (data, context) => {
    console.log('messages created event fired')
    console.log(app.channel(`convo/${data.conversationId}`).connections)
    console.log(app.channel('anonymous').connections)    // Shows identical connections as the line above
    return [app.channel('anonymous'), app.channel(`convo/${data.conversationId}`)];
  });
```

Because the client code receives these events when published over the anonymous channel, it leads me to believe this is a server-side issue.

**Module versions** (especially the part that's not working):

**NodeJS version**: v10.16.3

**Operating System**: Mac OS X 10_12_2

**Browser Version**: Chrome/77.0.3865.120

**Module Loader**: Webpack

**Comment by wgley3** [comment]
After debugging further, there is a change in behavior for joining channels from v3 to v4. This is the code we were using before:

```
app.service('conversations').on('created', (conversation, context) => {
    const { connection } = context.params;
    if (connection) {
      app.channel(`convo/${context.result._id}`).join(connection);
    }
  });
```
With the new v4 code, you can apparently no longer join the connection on a service event through `context.params.connection`. So instead, we changed this to join by filtering through the connections in `app.channel('users').connections` to find the correct active connection, then joining that to the channel.

Currently, if you join `context.params.connection` into the channel, it doesn't throw an error. Further, if you log `app.channel(..).connections` after joining with this method, the result is indistinguishable from the result if you do the correct method.

**Comment by daffl** [comment]
This should not have changed and may be a bug (potentially related to #1629)

**Comment by panstromek** [comment]
I got the same problem. If I join a channel with `params.connection`, connection is added but user doesn't receive any events. If I publish them to `'authentificated'` channel in `on('login')`, they are published correctly.

If I try to log connections from both channels, they look the same, but they are different objects (compared with strict eq `===`), so it seems like the connection object is copied somewhere on the way.

**Comment by daffl** [comment]
This has been fixed in [v4.3.11](https://github.com/feathersjs/feathers/releases/tag/v4.3.11).
